jgiven-core/src/main/java/com/tngtech/jgiven/report/asciidoc/AsciiDocReportGenerator.java

```diff
@@ -1,196 +1,164 @@
+    private final List<String> featureFiles = new ArrayList<>();
+    private final List<String> failedScenarioFiles = new ArrayList<>();
+    private final List<String> pendingScenarioFiles = new ArrayList<>();
+    private final AsciiDocReportBlockConverter blockConverter;
+    private File featuresDir;
+    private final Comparator<ReportModelFile> byFeatureName = Comparator.comparing(
+            modelFile -> null != modelFile.model.getName() ? modelFile.model.getName() : modelFile.model.getClassName());
 
-        allFiles.add( targetFileName );
-        if( !config.getTargetDir().exists() ) {
-            config.getTargetDir().mkdirs();
-        }
-        File targetFile = new File( config.getTargetDir(), targetFileName );
-        PrintWriter printWriter = PrintWriterUtil.getPrintWriter( targetFile );
+    public AsciiDocReportGenerator() {
+        blockConverter = new AsciiDocReportBlockConverter();
+    }
 
-        try {
-            new AbstractReportModelHandler().handle( model, new AsciiDocReportModelVisitor( printWriter ) );
-        } finally {
-            ResourceUtil.close( printWriter );
-        }
+    public AbstractReportConfig createReportConfig(String... args) {
+        return new AsciiDocReportConfig(args);
     }
 
-    private void generateIndexFile() {
-        PrintWriter printWriter = PrintWriterUtil.getPrintWriter( new File( config.getTargetDir(), "index.asciidoc" ) );
-        try {
-            printWriter.println( "= JGiven Documentation =\n" );
-            printWriter.println( ":toc: left\n" );
-            printWriter.println( "== Scenarios ==\n" );
-            printWriter.println( "=== Classes ===\n" );
-            printWriter.println( "include::allClasses.asciidoc[]" );
-        } finally {
-            ResourceUtil.close( printWriter );
+    public void generate() {
+        File targetDir = config.getTargetDir();
+        if (!targetDir.exists() && !targetDir.mkdirs()) {
+            log.error("Could not ensure target directory exists {}", targetDir);
+            return;
         }
-
-        printWriter = PrintWriterUtil.getPrintWriter( new File( config.getTargetDir(), "allClasses.asciidoc" ) );
-        try {
-            for( String fileName : allFiles ) {
-                printWriter.println( "include::" + fileName + "[]\n" );
-            }
-        } finally {
-            ResourceUtil.close( printWriter );
-
+        featuresDir = new File(targetDir.getPath() + "/features");
+        if (!featuresDir.exists() && !featuresDir.mkdirs()) {
+            log.error("Could not ensure feature directory exists {}", featuresDir);
+            return;
         }
 
-    }
+        completeReportModel.getAllReportModels().stream().sorted(byFeatureName).forEach(
+                reportModelFile -> writeFeatureToFile(reportModelFile.model, reportModelFile.file,
+                        completeReportModel.getStatistics(reportModelFile)));
```

Comment:
I don't like the format. I would prefer a line break before _.sorted_ and _.forEach_. It makes the code clearer and more more readable.